client/web/src/components/useUserHistory.ts

```diff
@@ -0,0 +1,117 @@
+    filePath?: string
+    lastAccessed: number
+}
+
+const LAST_REPO_ACCESS_FILEPATH = 'sourcegraph-last-repo-access.timestamp'
+const LOCAL_STORAGE_KEY = 'user-history'
+/** Maximum number of browser history entries to persist in local storage */
+const MAX_LOCAL_STORAGE_COUNT = 100
+
+/**
+ * Collects all browser history events and stores which repos/files are visited
+ * in local storage.  In the future, we should consider storing this history
+ * remotely in temporary settings (or similar). The history is used to
+ * personalize ranking in the fuzzy finder, but could theorically power other
+ * features like improve ranking in the search bar suggestions.
+ */
+export class UserHistory {
+    private repos: Map<string, Map<string, number>> = new Map()
+    private storage = window.localStorage
+    constructor() {
+        for (const entry of this.loadEntries()) {
+            this.onEntry(entry)
+        }
+    }
+    private saveEntries(entries: UserHistoryEntry[]): void {
+        entries.sort((a, b) => b.lastAccessed - a.lastAccessed)
+        const truncated = entries.slice(0, MAX_LOCAL_STORAGE_COUNT)
+        this.storage.setItem(LOCAL_STORAGE_KEY, JSON.stringify(truncated))
+        for (let index = MAX_LOCAL_STORAGE_COUNT; index < entries.length; index++) {
+            // Synchronize persisted entries with in-memory entries so that
+            // reloading the page doesn't change which entries are available.
+            this.deleteEntry(entries[index])
+        }
+    }
+    private loadEntries(): UserHistoryEntry[] {
+        return JSON.parse(this.storage.getItem(LOCAL_STORAGE_KEY) ?? '[]')
+    }
+    private deleteEntry(entry: UserHistoryEntry): void {
+        if (!entry.filePath) {
+            return
+        }
+        const repo = this.repos.get(entry.repoName)
+        if (!repo) {
+            return
+        }
+        repo.delete(entry.filePath)
+    }
+    public onEntry(entry: UserHistoryEntry): void {
+        let repo = this.repos.get(entry.repoName)
+        if (!repo) {
+            repo = new Map()
+            this.repos.set(entry.repoName, repo)
+        }
+        repo.set(LAST_REPO_ACCESS_FILEPATH, entry.lastAccessed)
+        if (!entry.filePath) {
+            return
+        }
+        repo.set(entry.filePath, entry.lastAccessed)
+    }
+    public persist(): void {
+        const entries: UserHistoryEntry[] = []
+        for (const repoName of this.repos.keys()) {
+            const repoMap = this.repos.get(repoName) ?? new Map<string, number>()
+            for (const filePath of repoMap.keys()) {
+                const lastAccessed = repoMap.get(filePath)
+                if (lastAccessed) {
+                    entries.push({ repoName, filePath, lastAccessed })
+                }
+            }
+        }
+        this.saveEntries(entries)
+    }
+    public onLocation(location: H.Location): boolean {
+        try {
+            const { repoName = '', filePath = '' } = parseBrowserRepoURL(
```

Comment:
```suggestion
            const { repoName, filePath } = parseBrowserRepoURL(
```
nit: we may not need default values here: [`repoName` is always `string`](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@1d5a32d19a770c129dfcfcbbcdecad698b60d653/-/blob/client/shared/src/util/url.ts?L138:1-148:32) and `filePath` [may be `undefined`](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@d7123f167db3710b531804fc4453c3861386b8a2/-/blob/client/web/src/components/useUserHistory.ts?L9).

Reply:
+1 Removed